app/src/main/java/edu/grinnell/kdic/MainActivity.java

```diff
@@ -114,99 +105,109 @@ protected void onStart() {
-                        hideVisualizeFragment();
-                        mBackStack.pop();
-                    }
-                    updateNavigationView();
-                }
-            }
-        };
+        View.OnClickListener onToggleVisualizeFragment = toggleVisualizeFragmentClickListener();
         mPlaybackToolbar.setNavigationOnClickListener(onToggleVisualizeFragment);
         mPlaybackToolbar.setOnClickListener(onToggleVisualizeFragment);
-        playPauseButton = (ImageView) findViewById(R.id.ib_play_pause);
-        playPauseButton.setOnClickListener(new View.OnClickListener() {
+        mPlayPauseButton = (ImageView) findViewById(R.id.ib_play_pause);
+        mPlayPauseButton.setOnClickListener(togglePlayPauseClickListener());
+    }
+
+    @NonNull
+    private View.OnClickListener togglePlayPauseClickListener() {
+        return new View.OnClickListener() {
             @Override
             public void onClick(View v) {
-                if (radioService.isPlaying()) {
+                if (mRadioService.isPlaying()) {
                     // pause
-                    radioService.pause();
+                    mRadioService.pause();
                     // switch to play icon
-                    playPauseButton.setImageResource(R.drawable.ic_play_arrow_white_24dp);
+                    mPlayPauseButton.setImageResource(R.drawable.ic_play_arrow_white_24dp);
                 } else {
                     // play
                     if (NetworkState.isOnline(MainActivity.this)) {
-                        if (!radioService.isLoading()) {
-                            if (radioService.isLoaded()) {
-                                playPauseButton.setImageResource(R.drawable.ic_pause_white_24dp);
-                                playPauseButton.clearAnimation();
+                        if (!mRadioService.isLoading()) {
+                            if (mRadioService.isLoaded()) {
+                                mPlayPauseButton.setImageResource(R.drawable.ic_pause_white_24dp);
+                                mPlayPauseButton.clearAnimation();
                             } else {
-                                playPauseButton.setImageResource(R.drawable.ic_loading_spinner);
+                                mPlayPauseButton.setImageResource(R.drawable.ic_loading_spinner);
 
                                 // rotation to use for loading icon
                                 RotateAnimation rotate;
-                                // different center point for rotation if playPauseButton is in the
+                                // different center point for rotation if mPlayPauseButton is in the
                                 // center of the screen
                                 if (mBackStack.peek() != R.id.visualizer)
                                     rotate = new RotateAnimation(0, 360,
                                             Animation.RELATIVE_TO_SELF, 0.5f, Animation.RELATIVE_TO_SELF,
```

Comment:
Use static imports. Also, handle these strange numbers appropriately by extracting them into meaningfully-named constants. eg. I have no idea what 360 or 0.5f means in this context.